features/toolbar/actions.js

```diff
@@ -46,15 +47,19 @@ export function hangup() {
                 .then(() => conference.leave());
         }
 
-        promise = promise
-            .then(() => dispatch(disposeLocalTracks()));
-
         if (connection) {
             promise = promise
                 .then(() => connection.disconnect());
         }
 
-        return promise;
+        // XXX Local tracks and local participant might exist without conference
+        // and connection initialized, so we need to explicitly clean them here.
+        // Furthermore, currently local tracks can be initialized before local
+        // participant is created, so we cannot hope that they will be destroyed
+        // when (and if) local participant leaves.
+        return promise
+            .then(() => dispatch(destroyLocalTracks()))
```

Comment:
Why do you dispatch destroyLocalTracks here? localParticipantLeft is dispatched bellow, which will result in PARTICIPANT_LEFT for the local participant. Shouldn't you reduce PARTICIPANT_LEFT in features/base/tracks to destroyLocalTracks if the participant who has left is the local participant?